wpiunits/src/main/java/edu/wpi/first/units/Measure.java

```diff
@@ -88,8 +88,11 @@ default Measure<U> times(double multiplier) {
   @SuppressWarnings("unchecked")
   default <U2 extends Unit<U2>> Measure<?> times(Measure<U2> other) {
     if (other.unit() instanceof Dimensionless) {
-      // scalar multiplication
+      // scalar multiplication of this
       return times(other.baseUnitMagnitude());
+    } else if (unit() instanceof Dimensionless) {
```

Comment:
Going through my tests for inverse and taking a `dimensionless` divided by a `velocity<dimensionless>` does not return a `time` but a `time per dimensionless` and it seems to be because of this check